code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -599,18 +599,49 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 
 	for (var/datum/reagent/R in reagent_list)
 		if (R.id == reagent)
-			if(!R.on_removal(amount))
-				return 0 //handled and reagent says fuck no
-			R.volume -= amount
-			update_total()
-			if(!safety)//So it does not handle reactions when it need not to
-				handle_reactions()
-			if(my_atom)
-				my_atom.on_reagent_change()
-			return 0
+			return remove_that_reagent(R, amount, safety)
+	return 1
+
+/datum/reagents/proc/remove_reagents(var/list/reagent_list, var/amount, var/safety)
+	if(!isnum(amount))
+		return 1
 
+	for(var/id in reagent_list)
+		if(has_reagent(id, amount))
+			remove_reagent(id, amount, safety)
 	return 1
 
+/datum/reagents/proc/remove_reagent_by_type(var/reagent_type, var/amount, var/safety)
+	if(!isnum(amount))
+		return 1
+
+	var/datum/reagent/R = get_reagent_by_type(type, amount)
+	if(R)
+		remove_that_reagent(R, amount, safety)
+
+/datum/reagents/proc/remove_reagents_by_type(var/list/reagent_types, var/amount, var/safety)
+	if(!isnum(amount))
+		return 1
+
+	for(var/datum/reagent/R in reagent_list)
+		if(is_type_in_list(R, reagent_types))
+			remove_that_reagent(R, amount, safety)
+
+/datum/reagents/proc/remove_that_reagent(var/datum/reagent/R, var/amount, var/safety)
```

Comment:
you need to update the readme with

1) your new remove_reagents which accepts a list
2) this thing, and explain how remove_that_reagent is different from remove_reagent

Reply:
chemistry-readme updated. It was a long time out of date.